lib/widgets/checkbox_widget.dart

```diff
@@ -0,0 +1,138 @@
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+import 'package:todo_app/screens/detail_page.dart';
+import 'package:todo_app/src/constants.dart';
+import 'package:todo_app/src/models/note.dart';
+import 'package:todo_app/src/provider/provider_data.dart';
+import 'package:todo_app/src/services.dart';
+
+///
+class CheckBoxWidget extends StatefulWidget {
+  ///
+  final Note note;
+
+  ///
+  const CheckBoxWidget({
+    Key? key,
+    required this.note,
+  }) : super(key: key);
+
+  @override
+  State<CheckBoxWidget> createState() => _CheckBoxState();
+}
+
+class _CheckBoxState extends State<CheckBoxWidget> {
+  List<Todo> taskList = [];
+
+  Future<List<Todo>> _getTask() async {
+    final _dataTasks =
+        Services.of(context).notesService.getTaskById(widget.note.id);
+    //
+
+    return _dataTasks;
+  }
+
+  @override
+  void initState() {
+    Future.microtask(
+      () => Provider.of<ProviderData>(context, listen: false)
+          .updateColor(widget.note.colorNote),
+    );
+    Provider.of<ProviderData>(context, listen: false).removeExit();
+    super.initState();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return FutureBuilder<List<Todo>>(
+      future: _getTask(),
+      builder: (context, snaphot) {
+        final _tasks = snaphot.data ?? [];
+        //taskList = _tasks;
+        context.watch<ProviderData>().taskListProvider = _tasks;
+        if (snaphot.hasData) {
+          return ListView.builder(
+            itemCount: context.watch<ProviderData>().taskListProvider.length,
+            itemBuilder: (context, index) {
+              return GestureDetector(
+                onLongPress: () {
+                  Services.of(context)
+                      .notesService
+                      .deleteTask(_tasks[index].id);
+                  // ignore: no-empty-block
+                  setState(() {});
+                },
+                child: Row(
+                  children: [
+                    Expanded(
+                      child: CheckboxListTile(
+                        secondary: RemoveTaskWidget(
+                          task: _tasks[index].id,
+                          taskIndex: index,
+                        ),
+                        checkColor:
+                            Provider.of<ProviderData>(context).getTextColor,
```

Comment:
Have a look into https://pub.dev/documentation/provider/latest/provider/Consumer-class.html
I think it will simplify this widget.

Also -- don't be afraid to add extra variables. `build` method is invoked in a single thread, synchronously. You don't need to keep fetching the Provider value each time, nothing's gonna happen to it.
```suggestion
                        checkColor:
                           providerData.getTextColor,
```